atomic_reactor/plugins/exit_koji_import.py

```diff
@@ -0,0 +1,257 @@
+        for platform in worker_metadatas:
+            for instance in worker_metadatas[platform]['output']:
+                instance['buildroot_id'] = platform + '-' + repr(instance['buildroot_id'])
+                outputs.append(instance)
+
+        return outputs
+
+    def get_buildroot(self, worker_metadatas):
+        """
+        Build the buildroot entry of the metadata.
+
+        :return: list, containing dicts of partial metadata
+        """
+        buildroots = []
+
+        # force ordering to work around python2 to python3 bug
+        platforms = []
+        for platform in worker_metadatas:
+            platforms.append(platform)
+        platforms.sort()
+        for platform in platforms:
+            for instance in worker_metadatas[platform]['buildroots']:
+                instance['id'] = platform + '-' + repr(instance['id'])
+                buildroots.append(instance)
+
+        return buildroots
+
+    def get_build(self, metadata, worker_metadatas):
+        start_time = int(atomic_reactor_start_time)
+
+        labels = df_parser(self.workflow.builder.df_path, workflow=self.workflow).labels
+
+        component = get_preferred_label(labels, 'com.redhat.component')
+        version = get_preferred_label(labels, 'version')
+        release = get_preferred_label(labels, 'release')
+
+        source = self.workflow.source
+        if not isinstance(source, GitSource):
+            raise RuntimeError('git source required')
+
+        extra = {'image': {'autorebuild': is_rebuild(self.workflow)}}
+        koji_task_id = metadata.get('labels', {}).get('koji-task-id')
+        if koji_task_id is not None:
+            self.log.info("build configuration created by Koji Task ID %s",
+                          koji_task_id)
+            try:
+                extra['container_koji_task_id'] = int(koji_task_id)
+            except ValueError:
+                self.log.error("invalid task ID %r", koji_task_id, exc_info=1)
+
+        fs_result = self.workflow.prebuild_results.get(AddFilesystemPlugin.key)
+        if fs_result is not None:
+            try:
+                fs_task_id = fs_result['filesystem-koji-task-id']
+            except KeyError:
+                self.log.error("%s: expected filesystem-koji-task-id in result",
+                               AddFilesystemPlugin.key)
+            else:
+                try:
+                    task_id = int(fs_task_id)
+                except ValueError:
+                    self.log.error("invalid task ID %r", fs_task_id, exc_info=1)
+                else:
+                    extra['filesystem_koji_task_id'] = task_id
+
+        help_metadata = {}
+        for platform in worker_metadatas:
+            worker_annotations = get_worker_build_info(self.workflow, platform)
+            if worker_annotations and 'help_file' in worker_annotations:
+                help_metadata['help'] = json.loads(worker_annotations['help_file'])
+        extra['image'].update(help_metadata)
```

Comment:
I think the problem that Tim is pointing out is that because the koji build is now being created from the orchestrator build,  the `'help'` value needs to be pulled from each worker build, and we're trying to combine those values into one.

In normal cases, the `'help'` value should be the same across all worker builds. But since each worker build will have its own binary buildroot image, there's the case in which for one platform the libraries needed for generating the help file are not installed. Tim's algorithm attempts to address that case.

Here's my understanding of what this block of code should do:

1. Only set the `'help'` key if any of the worker builds have set it
2. If multiple worker builds have set it, use any of the values; but the value of `None` should only be used if all worker builds have set it to `None`. In other words, a `str` value takes precedence over `None`.
3. The `'help'` key in orchestrator build will have a single value for help file (`str`, not a `list`)

I think the only thing needing change from Tim's suggestion is how `all_annotations` is computed. This might do it:
```python
    all_annotations = [get_worker_buildinfo(self.workflow, platform) for platform in worker_metadatas]
    help_known = ['help' in annotations for annotations in all_annotations]
    # Only set the 'help' key when any 'help_file' annotation is set
    if any(help_known):
        # See if any are not None
        for known, annotations in zip(help_known, all_annotations):
            if known:
                if annotations['help'] is not None:
                    metadata['help'] = annotatations['help']
                    break
        else:
            # They are all None
            metadata['help'] = None
```

Note: I think this whole block should be in its own method.